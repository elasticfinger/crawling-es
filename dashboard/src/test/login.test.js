import LoginContainer from '../containers/auths/Login'
import React from 'react'
import { render ,fireEvent} from '@testing-library/react'


test('Login snapshot' , ()=>{
    const utils = render(<LoginContainer/>)
    expect(utils.container).toMatchSnapshot()
})

test('Login input id', ()=>{
    const {container} =render(<LoginContainer/>)
    const input_id = container.querySelectorAll('input')[0];
    const event = {
        target : {
            value : "hello",
            name : "name"
        }
    }
    fireEvent.change(input_id,event)
    expect(input_id).toHaveValue('hello');
    expect(input_id).not.toHaveValue('');
})

test('login input password',()=>{
    const {container} =render (<LoginContainer/>)
    const input_password = container.querySelectorAll("input")[1];
    const event = {
        target :{
            value : "world",
            name : "password"
        }
    }
    fireEvent.change(input_password, event)
    expect(input_password).toHaveValue('world');
    expect(input_password).not.toHaveValue('');
})