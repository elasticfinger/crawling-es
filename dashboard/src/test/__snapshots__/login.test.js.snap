// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login snapshot 1`] = `
<div>
  <input
    class="sc-AxirZ jfUsbt"
    name="name"
    placeholder="아이디"
    type="text"
    value=""
  />
  <input
    class="sc-AxirZ jfUsbt"
    name="password"
    placeholder="비밀번호"
    type="password"
    value=""
  />
  <button
    class="sc-AxjAm kXwayF"
  >
    로 그 인
  </button>
  <button
    class="sc-AxjAm kXwayF"
  >
    회원가입
  </button>
</div>
`;
